exporter/azuremonitorexporter/traceexporter.go

```diff
@@ -0,0 +1,660 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuremonitorexporter
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights/contracts"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"google.golang.org/grpc/codes"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+const (
+	spanAttributeKeyComponent = "component"
+
+	// common attributes keys for HTTP spans
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#common-attributes
+	spanAttributeKeyHTTPMethod     = "http.method"
+	spanAttributeKeyHTTPUrl        = "http.url"
+	spanAttributeKeyHTTPTarget     = "http.target"
+	spanAttributeKeyHTTPHost       = "http.host"
+	spanAttributeKeyHTTPScheme     = "http.scheme"
+	spanAttributeKeyHTTPStatusCode = "http.status_code"
+	spanAttributeKeyHTTPFlavor     = "http.flavor"
+
+	// attributes key for HTTP server spans
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#http-server
+	spanAttributeKeyHTTPServerName = "http.server_name"
+	spanAttributeKeyHostName       = "host.name"
+	spanAttributeKeyHostPort       = "host.port"
+	spanAttributeKeyHTTPRoute      = "http.route"
+	spanAttributeKeyHTTPClientIP   = "http.client_ip"
+
+	// general purpose peer attribute keys used across various types
+	spanAttributeKeyPeerAddress  = "peer.address"
+	spanAttributeKeyPeerService  = "peer.service"
+	spanAttributeKeyPeerHostname = "peer.hostname"
+	spanAttributeKeyPeerPort     = "peer.port"
+	spanAttributeKeyPeerIP       = "peer.ip"
+	spanAttributeKeyPeerIPv4     = "peer.ipv4"
+	spanAttributeKeyPeerIPv6     = "peer.ipv6"
+
+	// RPC attribute keys
+	spanAttributeKeyRPCStatusCode    = "status_code"
+	spanAttributeKeyRPCStatusMessage = "status_message"
+
+	// Database keys
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-database.md
+	spanAttributeKeyDbType      = "db.type"
+	spanAttributeKeyDbInstance  = "db.instance"
+	spanAttributeKeyDbStatement = "db.statement"
+	spanAttributeKeyDbUser      = "db.user"
+)
+
+type traceExporter struct {
+	config           *Config
+	transportChannel transportChannel
+}
+
+func toHex(source []byte) string {
+	if source == nil {
+		return ""
+	}
+
+	return fmt.Sprintf("%02x", source)
+}
+
+func formatParentChild(parent string, child string) string {
```

Comment:
@Dmitry-Matveev must have already pushed a change in our endpoint that wouldn't require this formatting.

@Dmitry-Matveev, please confirm if it's in prod already